internal/pgbackrest/pgbackrest.go

```diff
@@ -60,22 +62,44 @@ func (exec Executor) StanzaCreateOrUpgrade(ctx context.Context, configHash strin
 		stanzaCmd = "upgrade"
 	}
 
+	var reposWithVolumes []v1beta1.PGBackRestRepo
+	for _, repo := range postgresCluster.Spec.Backups.PGBackRest.Repos {
+		if repo.Volume != nil {
+			reposWithVolumes = append(reposWithVolumes, repo)
+		}
+	}
+
+	grep := "grep %s-path /etc/pgbackrest/conf.d/pgbackrest_instance.conf"
+
+	var checkRepoCmd string
+	if len(reposWithVolumes) > 0 {
+		repo := reposWithVolumes[0]
+		checkRepoCmd = checkRepoCmd + fmt.Sprintf(grep, repo.Name)
+
+		reposWithVolumes = reposWithVolumes[1:]
+		for _, repo := range reposWithVolumes {
+			checkRepoCmd = checkRepoCmd + fmt.Sprintf(" && "+grep, repo.Name)
+		}
+	}
+
 	// this is the script that is run to create a stanza.  First it checks the
 	// "config-hash" file to ensure all configuration changes (e.g. from ConfigMaps) have
 	// propagated to the container, and if not, it prints an error and returns with exit code 1).
 	// Otherwise, it runs the pgbackrest command, which will either be "stanza-create" or
 	// "stanza-upgrade", depending on the value of the boolean "upgrade" parameter.
 	const script = `
-declare -r hash="$1" stanza="$2" message="$3" cmd="$4"
+declare -r hash="$1" stanza="$2" message="$3" cmd="$4" check_repo_cmd="$5"
 if [[ "$(< /etc/pgbackrest/conf.d/config-hash)" != "${hash}" ]]; then
     printf >&2 "%s" "${message}"; exit 1;
+elif ! bash -c "${check_repo_cmd}"; then
+		printf >&2 "%s" "${message}"; exit 1;
```

Comment:
🤔 Should we consider another/different error message here?  The current message specifically specifically references a "config hash mismatch", which really isn't applicable this new check.

At the very least, prob worth adding adding a note/comment in there around the reasoning for leveraging that same error message, e.g. to sure we're not confused with other hashing logic in the future.

Reply:
Good idea 👍